builtin/providers/aws/resource_aws_instance.go

```diff
@@ -760,6 +760,10 @@ func fetchRootDeviceName(ami string, conn *ec2.EC2) (*string, error) {
 		rootDeviceName = image.BlockDeviceMappings[0].DeviceName
 	}
 
+	if rootDeviceName == nil {
+		return nil, fmt.Errorf("[WARN] Error finding Root Device Name for AMI (%s)", ami)
```

Comment:
In the linked issue the problem was a bad AMI ID, which means we would have hit the early nil return up on L735.

We specifically added that "no error when AMI not found" thingy in https://github.com/hashicorp/terraform/pull/1250 to fix unrecoverable error states because it was being hit in `Read`.

So we need to figure out how we can maintain the "allow users to correct bad AMIs" while still getting a specific error message when they specify one.